clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -400,7 +400,7 @@ public long timeToNextPoll(long now) {
 
         log.debug("Performing assignment using strategy {} with subscriptions {}", assignor.name(), subscriptions);
 
-        Map<String, Assignment> assignment = assignor.assign(metadata.fetch(), subscriptions);
+        Map<String, Assignment> assignment = assignor.assign(metadata.fetch(), subscriptions, latestGeneration().generationId);
```

Comment:
This might be a little brittle since it depends on the `generation` field getting updated before the rebalance has completed. Perhaps we can pass `generation` as an argument to `performAssignment`?

Reply:
That makes sense. Added the `generation` argument. Thanks!